### DISCUSSION TITLE
How can is reset specific field inside a form?

### DISCUSSION THREAD

**Discussion by amirgc** [danvick/flutter_form_builder#746] [Q&A] [answered]
I want to reset the dropdown1 value on change of another dropdown2 so i can load new list on dropdown 1. but i want to reset the dropdown1 value before that.

I want to do this because form is maintaining the value(state) of previous select. And when i change the value on dropdown2 then new list come up, there is no matching value on the new list which matches the previously maintained state of dropdown and i get the below error.
```
There should be exactly one item with [DropdownButton]'s value: 2565. 
Either zero or 2 or more [DropdownMenuItem]s were detected with the same value
'package:flutter/src/material/dropdown.dart':
Failed assertion: line 834 pos 15: 'items == null || items.isEmpty || value == null ||
              items.where((DropdownMenuItem<T> item) {
                return item.value == value;
              }).length == 1'
```

If there is way to solve this , then that will be more helpful too.

**Comment by danvick** [comment, marked as answer]
You can always add a key to the field then use it to reset the field. Like so:
```dart
// Create key
final _fieldKey = GlobalKey<FormBuilderFieldState>();
...

FormBuilderDropdown(
  // Add key to field widget
  key: _fieldKey,
  ...
)

...

// Reset field using key
_fieldKey.currentState.reset();
```
